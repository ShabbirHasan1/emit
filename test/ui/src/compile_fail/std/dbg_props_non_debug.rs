struct Data {
    id: i32,
}

fn main() {
    emit::dbg!(a: Data { id: 42 });
}
