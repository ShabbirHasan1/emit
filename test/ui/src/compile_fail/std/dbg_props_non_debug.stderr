error[E0277]: capturing with `#[emit::as_debug]` requires `Data` implements `Debug`. Use another of the `#[emit::as_*]` attributes to capture this value using a trait it does implement.
 --> src/compile_fail/std/dbg_props_non_debug.rs:6:16
  |
6 |     emit::dbg!(a: Data { id: 42 });
  |                ^^^^^^^^^^^^^^^^^^ the trait `std::fmt::Debug` is not implemented for `Data`
  |
  = help: the trait `CaptureAsAnonDebug` is implemented for `str`
  = note: required for `Data` to implement `CaptureAsAnonDebug`
note: required by a bound in `__private_capture_anon_as_debug`
 --> $WORKSPACE/src/macro_hooks.rs
  |
  |     fn __private_capture_anon_as_debug(&self) -> Option<Value<'_>>
  |        ------------------------------- required by a bound in this associated function
  |     where
  |         Self: CaptureAsAnonDebug,
  |               ^^^^^^^^^^^^^^^^^^ required by this bound in `__PrivateCaptureHook::__private_capture_anon_as_debug`
help: consider annotating `Data` with `#[derive(Debug)]`
  |
  1 + #[derive(Debug)]
  2 | struct Data {
    |
